### ISSUE TITLE
Vocabulary for English and German

### ISSUE THREAD

**Issue by abhudev** [vineetm/ell-881-2018-deep-learning#17]
It is specified that vocabulary should be 30,000 most frequent words from the data, but for En-De translation, we will need different vocabularies. So should we keep the same number of words in German vocabulary as well? Also, am I correct in assuming the construction of vocabularies:

Source vocab: By combining all English data from all three datasets
Target vocab for NMT in German: From German data
Target vocab for parsing: from the parse trees

**Comment by vineetm** [comment]
@abhudev Thanks for your question. Yes, for any parameters not explicitly specified in the [Project description](https://github.com/vineetm/ell-881-2018-deep-learning/blob/master/project1/project1.md), use the parameters specified in the paper. There is a nice summary available int the [paper github](https://github.com/Maluuba/gensen/blob/master/example_config.json) 

So yes, the target vocab is 30,000

You are absolutely correct in your assumptions about vocabularies. Since we have the same RNN as source in all the three models, the source vocab is constructed by combining all the English data.

**Comment by abhudev** [comment]
@vineetm Thank you for the clarification Sir!

**Comment by abhudev** [comment]
Also, to clarify, should the total size of the vocabulary be 30,000, including the UNK and PAD tokens, or only counting the words got from the training data?

**Comment by vineetm** [comment]
Yes, a total of 30K including UNK and pad tokens.

**Comment by anshumitts** [comment]
@vineetm Why should we include PAD tokens in the vocabulary, isn't it just embeddings of padding zeros?

**Comment by vineetm** [comment]
@anshumitts We discussed this in class. You would feed a vector for each word in your input/output. When you batch words you **PAD** the short sentences. You would still require to index the PAD token to get a token index.

Zero embedding business will come into play, when you explicitly set the row corresponding to PAD token as zero. But you will still need the PAD token in your vocabulary.

Let me know if the above explanation helps.
